openverse_catalog/dags/providers/provider_api_scripts/nypl.py

```diff
@@ -1,210 +1,229 @@
 import logging
+import re
 from urllib.parse import parse_qs, urlparse
 
 from airflow.models import Variable
 from common.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.image import ImageStore
+from common.loader.provider_details import ImageCategory
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
 logging.basicConfig(
     format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
 )
 logger = logging.getLogger(__name__)
 
-LIMIT = 500
-DELAY = 1.0
-RETRIES = 3
-PROVIDER = prov.NYPL_DEFAULT_PROVIDER
-BASE_ENDPOINT = "http://api.repo.nypl.org/api/v1/items/search"
-METADATA_ENDPOINT = "http://api.repo.nypl.org/api/v1/items/item_details/"
 NYPL_API = Variable.get("API_KEY_NYPL", default_var=None)
-TOKEN = f"Token token={NYPL_API}"
+FILETYPE_PATTERN = r" .(jpeg|gif) "
+
+
+class NyplDataIngester(ProviderDataIngester):
+    providers = {"image": prov.NYPL_DEFAULT_PROVIDER}
+    headers = {"Authorization": f"Token token={NYPL_API}"}
+    endpoint = "http://api.repo.nypl.org/api/v1/items/search"
+    metadata_endpoint = "http://api.repo.nypl.org/api/v1/items/item_details/"
```

Comment:
Should this maybe be the following for reuseability?
```suggestion
    endpoint = "http://api.repo.nypl.org/api/v1/items"
    search_endpoint = f"{endpoint}/search"
    metadata_endpoint = f"{endpoint}/item_details/"
```